### ISSUE TITLE
Updating to 4.3.2 from 4.0.3 seems to have broken my DB access to MySQL (MariaDB)

### ISSUE THREAD

**Issue by tobogdan** [LycheeOrg/Lychee#1018]
### Detailed description of the problem
When initiating the update process (following the posted Lychee Upgrade guide), and going from 4.0.3 to 4.3.2 resulted in my system being unable to parse the albums after restarting the service. 

Funnily enough, it worked immediately after the update for a few minutes, until I did 'npm run compile' to change the size of two elements in the splash page, which I assume also restarted the session. Note that this is running on my old 32-bit install where I had filed an issue earlier and we had resolved the database issue by manually updating my sub-albums to have the correct/truncated 32-bit IDs. I had verified that it wasn't happening again by manually checking my DB and confirming it looked correct.

Using the browser's debug tools, I found that it's having issues enumerating the albums:
![image](https://user-images.githubusercontent.com/2244654/120081269-d8c8f300-c08a-11eb-9d51-c0384d5e42c0.png)

I'm also having issues logging in to the admin account at all, with the generic "Server Error or API Not Found" message:
![image](https://user-images.githubusercontent.com/2244654/120081348-407f3e00-c08b-11eb-90ad-341f6c90ca86.png)

Running diagnostics only lists an error and a few warnings that don't seem to be causing this level of breakage:
![image](https://user-images.githubusercontent.com/2244654/120081376-5b51b280-c08b-11eb-8d19-b85ec5d7fc49.png)

Any thoughts or things to try would be appreciated.

### Steps to reproduce the issue
**Steps to reproduce the behavior:**
1. Grab an older build (around 4.0.3 or so), and set up some albums, using a 32-bit configuration for the whole stack (32-bit may not be necessary)
2. Once you've confirmed everything is up and running, including sub-albums, perform the upgrade steps as outlined in the official documentation (https://lycheeorg.github.io/docs/update.html)
3. Verify the update worked by logging back in to the site and seeing functionality is still working
4. Make some arbitrary changes to the splash page (i.e., change public/dist/landing.css) and run npm run compile
5. Observe that the site is now down and unable to connect with the issues described previously
6. Pull your hair out thinking you screwed up something obvious and lose days poring over configuration and DB connection strings, confirming that even config/database.php is perfectly all right numerous times.

### Output of the diagnostics  [REQUIRED]
Diagnostics
-----------
Error: Wrong property for image_overlay_type in database, expected exif or desc or date or none, got takedate
Warning: Dropbox import not working. dropbox_key is empty.
Warning: You may experience problems when uploading a photo of large size. Take a look in the FAQ for details.
Warning: You may experience problems when uploading a photos of large size. Take a look in the FAQ for details.
Warning: You may experience problems when uploading a large amount of photos. Take a look in the FAQ for details.
Info: Latest version of PHP is 8
Warning: Using 32 bit PHP, recommended upgrade to 64 bit

System Information
------------------
Lychee Version (git):            master (75c78e5) - Data not in Cache
DB Version:                      4.3.2

composer install:                --no-dev
APP_ENV:                         production
APP_DEBUG:                       false

System:                          Linux
PHP Version:                     7.4
Max uploaded file size:          2M
Max post size:                   8M
MySQL Version:                   10.3.29-MariaDB-0ubuntu0.20.04.1

Imagick:                         1
Imagick Active:                  1
Imagick Version:                 1690
GD Version:                      2.2.5

Config Information
------------------
version:                         040302
check_for_updates:               1
sorting_Photos_col:              title
sorting_Photos_order:            ASC
sorting_Albums_col:              description
sorting_Albums_order:            ASC
imagick:                         1
skip_duplicates:                 0
small_max_width:                 0
small_max_height:                360
medium_max_width:                1920
medium_max_height:               1080
lang:                            en
layout:                          1
image_overlay_type:              takedate
default_license:                 reserved
compression_quality:             90
full_photo:                      1
delete_imported:                 1
Mod_Frame:                       0
Mod_Frame_refresh:               30
landing_page_enable:             1
landing_owner:                   <<REMOVED>>
landing_title:                   <<REMOVED>>
landing_subtitle:                <<REMOVED>>
landing_facebook:
landing_flickr:
landing_twitter:
landing_instagram:               <<REMOVED>>
landing_youtube:                 <<REMOVED>>
landing_background:              dist/landing.jpg
thumb_2x:                        1
small_2x:                        0
medium_2x:                       0
site_title:                      <<REMOVED>>
site_copyright_enable:           1
site_copyright_begin:            <<REMOVED>>
site_copyright_end:              <<REMOVED>>
additional_footer_text:
display_social_in_gallery:       0
public_search:                   0
SL_enable:                       0
SL_for_admin:                    0
public_recent:                   0
recent_age:                      1
public_starred:                  0
downloadable:                    0
photos_wraparound:               1
map_display:                     0
zip64:                           1
map_display_public:              0
map_provider:                    Wikimedia
force_32bit_ids:                 0
map_include_subalbums:           0
update_check_every_days:         3
has_exiftool:                    0
share_button_visible:            0
import_via_symlink:              0
has_ffmpeg:                      1
location_decoding:               0
location_decoding_timeout:       30
location_show:                   1
location_show_public:            0
rss_enable:                      0
rss_recent_days:                 7
rss_max_items:                   100
prefer_available_xmp_metadata:   0
editor_enabled:                  1
lossless_optimization:           0
swipe_tolerance_x:               150
swipe_tolerance_y:               250
local_takestamp_video_formats:   .avi|.mov
log_max_num_line:                1000
unlock_password_photos_with_url_param: 0
nsfw_visible:                    1
nsfw_blur:                       0
nsfw_warning:                    0
nsfw_warning_admin:              0
map_display_direction:           1
album_subtitle_type:             oldstyle
upload_processing_limit:         4
public_photos_hidden:            1

### Browser and system
Any browser shows this error (Chrome and Firefox most thoroughly tested.
Server is running on Ubuntu Armbian armhf (32b) architecture, on 21.05.2 stable (Focal Fossa)

**Comment by kamil4** [comment]
Your first issue (`album.thumbs`) should have been fixed by https://github.com/LycheeOrg/Lychee-front/pull/257 I think, but we haven't made a release since... If you are eager to give it a try, you will need to use our development tree on github.

There have been changes since 4.0.4 to how the admin account is handled internally. Perhaps something went wrong during the database migration in your case as you were upgrading to 4.3.2; have you tried simply resetting the admin login/password as described in https://lycheeorg.github.io/docs/faq.html#how-to-reset-username-and-password ?

**Comment by tobogdan** [comment]
Thanks for the response - I have been poking at it this past hour and have found that there's actually an issue in the update sequence - if you are using a custom frontend like I am, the instructions in the update steps don't include that you need to do a
`git submodule update` step, to update Lychee-front and the build targets.
Missing this step meant that I was still using frontend version 4.0.1 (as opposed to 4.0.10 as reported in the json file now).

Updating that seems to have fixed my db access and the admin login.

However, it seems to have broken my ability to update the dists source files - when I make changes to get to the slight modifications I am wanting to get from my past install, every time I run `npm run compile`, it resets the source CSS files and reverts my changes.

I'll update this issue with my findings for the last issue once I get there, but I think at this point, an update to the documentation should happen to just remind the user that if they've done any customizations to the submodules, they should manually call that submodule update step as well.

**Comment by ildyria** [comment]
> it resets the source CSS files and reverts my changes.

that is why there is the `user.css` which is not overwritten and you can add your own configurations.

**Comment by tobogdan** [comment]
@ildyria, thanks for the heads up. I missed that outright in the notes - it doesn't mention anywhere I looked that it has been changed over. It's been a long time since I checked on the project, to be honest, so it may have been me that missed it outright. 

I think that covers all the issues I was having, I will close this ticket for now, and hope that documentation updates will cover these cases as the project evolves :)
